crates/llvm-context/src/polkavm/context/function/runtime/immutable_data_load.rs

```diff
@@ -111,6 +122,10 @@ where
         context.set_basic_block(return_block);
         context.build_return(None);
 
+        if let Some(dinfo) = context.debug_info() {
```

Comment:
This is inherently tied to `fn set_current_function`.

Whenever we do this for a function we have an inconsistency. Because we set the current function, push a di scope, pop a di scope, but do not unset the current function.

Which implies that at any point, we assume to have a function set (because there is no way to unset it).

Which implies that this can and should be done in `fn set_current_function` too, before pushing the new scope.

(This and the other comment should apply for functions in general)

Reply:
I mean if this doesn't work. I might be missing something. But if we mismatch the code we emit with the function scope we emit, it smells like something should be broken. WDYT?

Reply:
Moved function debug-info generation into `set_current_function`.

Reply:
Ok that might not work at all. We might have to stick with the implication that we are done at the end here.

Reply:
> Ok that might not work at all. We might have to stick with the implication that we are done at the end here.

(This was addressing my own comment I wrote earlier above)

> Moved function debug-info generation into set_current_function.

Sounds good. Assuming this has to be done anyways when setting the function, seems like the right place :) I guess popping the context probably needs to inherently stay separate. For now I'm fine with sprinkling `context.pop_debug_scope();` everywhere we expect to not come back emitting more code.